core/src/utils/periodic_task_test.cpp

```diff
@@ -418,4 +418,39 @@ UTEST_F(PeriodicTaskLog, ErrorLog) {
     task.Stop();
 }
 
+UTEST(PeriodicTask, EnableDisable) {
```

Comment:
Add a test that checks the task does not keep running periodically if `ForceStepAsync` is run while `enabled=false`

Reply:
This test will also check that `ForceStepAsync` ignores `enabled`